pandas/tseries/offsets.py

```diff
@@ -299,8 +299,53 @@ def is_on_offset(self, dt):
         # TODO, see #1395
         return True
 
+    def _repr_attrs(self) -> str:
+        # The DateOffset class differs from other classes in that members
```

Comment:
this seems like it could be integrated to the other classes no? surely a follown

Reply:
yah, there are a lot of `__dict__`-related workarounds that can be cleaned up once _everything_ is a cdef class